### PR TITLE
Add cluster class patch for kubevip control plane endpoint 

### PR DESCRIPTION
**Pull request by thunderboltsid**
Prepend sed command on KCPT prekubadm commands list. The sed commands mutate the kube-vip template with variables to configure kube-vip.

### CODE REVIEW

File: templates/clusterclass/kcpt.yaml [hunk 1]
```diff
@@ -40,9 +40,9 @@ spec:
                       - name: vip_arp
                         value: "true"
                       - name: address
-                        value: "${CONTROL_PLANE_ENDPOINT_IP}"
+                        value: "control_plane_endpoint_ip"
                       - name: port
-                        value: "${CONTROL_PLANE_ENDPOINT_PORT=6443}"
+                        value: "control_plane_endpoint_port"
```
Comment (deepakm-ntnx) [inline templates/clusterclass/kcpt.yaml:45]: nit: pls file a tracker to add them as vars or default the values on line 63, 65 and 67 below

Comment (deepakm-ntnx) [reply templates/clusterclass/kcpt.yaml:45]: Also as discussed in future we may have to make the entire section as variable if user want to update any other variables which we are not sed-ing yet

**Review by deepakm-ntnx — DISMISSED** [review]

**Review by deepakm-ntnx — APPROVED** [review]

File: templates/clusterclass/kcpt.yaml (hunk 1 above)
Comment (deepakm-ntnx) [reply templates/clusterclass/kcpt.yaml:45]: as discussed earlier, we will add later in separate PR if needed

File: templates/cluster-template-clusterclass.yaml [hunk 2]
```diff
@@ -108,6 +108,22 @@ spec:
         kind: NutanixClusterTemplate
         matchResources:
           infrastructureCluster: true
+    - jsonPatches:
```
Comment (dkoshkin) [inline templates/cluster-template-clusterclass.yaml:111]: There are other envs in the template `KUBEVIP_SVC_ENABLE`, `KUBEVIP_LB_ENABLE` , etc

What do you think about adding a patch with the whole content of this file?
This way we can still have direct variable substation and allow users to override the whole thing if they need to change other parts of it. 

It also has the benefit of keeping the templates clean and reusable in other usecases

**Review by dkoshkin — CHANGES_REQUESTED** [review]

File: templates/cluster-template-clusterclass.yaml (hunk 2 above)
Comment (faiq) [reply templates/cluster-template-clusterclass.yaml:111]: I agree. Also I don't understand why its being done with sed in a preKubeadmCommnd

Comment (deepakm-ntnx) [reply templates/cluster-template-clusterclass.yaml:111]: already tracked thru https://jira.nutanix.com/browse/KRBN-8114 regarding extra parameter related changes

**Review by dkoshkin — APPROVED** [review]
Approving these changes, but lets please come back it for https://jira.nutanix.com/browse/KRBN-8114

**Pull request merged**